pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -487,6 +408,278 @@ def phase_str(phi: float) -> str:
+        """Formats a phase value for printing."""
+        value = (((phi + np.pi) % (2 * np.pi)) - np.pi) / np.pi
+        if value == -1:
+            return r"$\pi$"
+        elif value == 0:
+            return "0"  # pragma: no cover - just for safety
+        else:
+            return rf"{value:.2g}$\pi$"
+
+    n_channels = len(seq.declared_channels)
+    if not n_channels:
+        raise RuntimeError("Can't draw an empty sequence.")
+
+    # Boxes for qubit and phase text
+    area_ph_box = dict(boxstyle="round", facecolor="ghostwhite", alpha=0.7)
+    q_box = dict(boxstyle="round", facecolor="orange")
+    ph_box = dict(boxstyle="round", facecolor="ghostwhite")
+
+    # Sample the sequence
+    sampled_seq = sample(seq)
+
+    # Draw masked register
+    if draw_register:
+        pos = np.array(seq.register._coords)
+        if isinstance(seq.register, Register3D):
+            labels = "xyz"
+            fig_reg, axes_reg = seq.register._initialize_fig_axes_projection(
+                pos,
+                blockade_radius=35,
+                draw_half_radius=True,
+            )
+            fig_reg.get_layout_engine().set(w_pad=6.5)
+
+            for ax_reg, (ix, iy) in zip(
+                axes_reg, combinations(np.arange(3), 2)
+            ):
+                seq.register._draw_2D(
+                    ax=ax_reg,
+                    pos=pos,
+                    ids=seq.register._ids,
+                    plane=(ix, iy),
+                    masked_qubits=seq._slm_mask_targets,
+                )
+                ax_reg.set_title(
+                    "Masked register projected onto\n the "
+                    + labels[ix]
+                    + labels[iy]
+                    + "-plane"
+                )
+
+        elif isinstance(seq.register, Register):
+            fig_reg, ax_reg = seq.register._initialize_fig_axes(
+                pos,
+                blockade_radius=35,
+                draw_half_radius=True,
+            )
+            seq.register._draw_2D(
+                ax=ax_reg,
+                pos=pos,
+                ids=seq.register._ids,
+                masked_qubits=seq._slm_mask_targets,
+            )
+            ax_reg.set_title("Masked register", pad=10)
+
+    (fig, ch_axes, data) = draw_samples(
+        sampled_seq,
+        sampling_rate,
+        draw_input,
+        draw_modulation,
+        draw_phase_curve,
+        draw_target_regions=False,
+    )
+
+    # Gather additional data
+    for ch, sch in seq._schedule.items():
+        interp_pts: defaultdict[str, list[list[float]]] = defaultdict(list)
+        target: dict[Union[str, tuple[int, int]], Any] = {}
+
+        for slot in sch:
+            if slot.ti == -1:
+                target["initial"] = slot.targets
+                continue
+            if slot.type == "target":
+                target[(slot.ti, slot.tf - 1)] = slot.targets
+                continue
+            if slot.type == "delay":
+                continue
```

Comment:
All of this is already taking place in `gather_data` and are stored in `ChannelDrawContent`.